### PR TITLE
update test for mageck/test (fka PR 4550)

### PR DESCRIPTION
**Pull request by LaurenceKuhl**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/mageck/test/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                            process.out.versions,
+                            file(process.out.gene_summary.get(0).get(1)).readLines()[0],
+                            file(process.out.sgrna_summary.get(0).get(1)).readLines()[0],
+                            file(process.out.r_script.get(0).get(1)).readLines()[0]
+                            ).match()
+                }
```
Comment (famosab) [inline modules/nf-core/mageck/test/tests/main.nf.test:35]:
```suggestion
                { assert snapshot(process.out.).match() }
```

File: modules/nf-core/mageck/test/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                            process.out.versions,
+                            file(process.out.gene_summary.get(0).get(1)[0]).readLines()[0],
+                            file(process.out.sgrna_summary.get(0).get(1)[0]).readLines()[0]
+                            ).match()
+                }
```
Comment (famosab) [inline modules/nf-core/mageck/test/tests/main.nf.test:63]:
```suggestion
                { assert snapshot(process.out).match() }
```

And this I think.

**Comment by LaurenceKuhl** [comment]
hi @SPPearce the conda test is failing and it's working for me locally, do you have any experience on potential discrepancies? :/

**Comment by SPPearce** [comment]
> hi @SPPearce the conda test is failing and it's working for me locally, do you have any experience on potential discrepancies? :/

Could be numerical machine precision differences on different architectures

**Comment by SPPearce** [comment]
I can't see anything obvious and can't seem to find the artefacts from running on github actions, but I suspect that it is machine precision issues. I would test some of the contents of those files instead.

**Pull request merged**